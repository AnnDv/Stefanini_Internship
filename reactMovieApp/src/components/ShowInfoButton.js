import React, { useState } from "react";
import MovieDescription from "./MovieDescription";
import CustomModal from "./CustomModal";
import './ShowInfoButton.css';
import PropTypes from 'prop-types';


function ShowInfoButton({item}) {
  const [isOpen, setIsOpen] = useState(false);

  const isOpenHandler = () => setIsOpen(true);
  const isCloseHandler = () => setIsOpen(false);

  return (
    <>
      <div className="card-body">
        <button className="button" onClick={isOpenHandler}>View More</button>
        <CustomModal open={isOpen} onClose={isCloseHandler}>
            <MovieDescription item={item}/>
        </CustomModal>
      </div>
    </>
    )
}

ShowInfoButton.propTypes = {
    item: PropTypes.object
}

export default ShowInfoButton;