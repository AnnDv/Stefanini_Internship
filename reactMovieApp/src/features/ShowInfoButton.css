@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@500&display=swap');

.card-body {
    padding: 0 10px;
    position: relative;
}

.button {
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);
    background-color: #688282;
    border: none;
    padding: 10px 15px;
    border-radius: 19px;
    color: #fff;
    font-size: 16px;
    font-family: 'Roboto', sans-serif;
}

.header {
    font-family: 'Roboto', sans-serif;
}