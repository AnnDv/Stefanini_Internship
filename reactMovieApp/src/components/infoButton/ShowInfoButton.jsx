import React, { useState } from 'react';
import PropTypes from 'prop-types';
import MovieDescription from '../../features/movieComponents/MovieDescription.jsx';
import CustomModal from './CustomModal.jsx';
import './ShowInfoButton.css';

function ShowInfoButton({ overview }) {
  const [isOpen, setIsOpen] = useState(false);

  const isOpenHandler = () => setIsOpen(true);
  const isCloseHandler = () => setIsOpen(false);

  return (
    <>
      <div className="card-body">
        <button className="button" onClick={isOpenHandler}>View More</button>
        <CustomModal open={isOpen} onClose={isCloseHandler}>
            <MovieDescription data-testid="showInfo" overview={overview}/>
        </CustomModal>
      </div>
    </>
  );
}

ShowInfoButton.propTypes = {
  overview: PropTypes.string,
};

export default ShowInfoButton;
