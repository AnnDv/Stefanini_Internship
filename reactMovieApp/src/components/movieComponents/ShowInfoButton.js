import React, { useState } from "react";
import MovieDescription from "./MovieDescription";
import CustomModal from "./CustomModal";
import './ShowInfoButton.css';
import PropTypes from 'prop-types';

function ShowInfoButton({overview}) {
  const [isOpen, setIsOpen] = useState(false);

  const isOpenHandler = () => setIsOpen(true);
  const isCloseHandler = () => setIsOpen(false);

  return (
    <>
      <div className="card-body">
        <button className="button" onClick={isOpenHandler}>View More</button>
        <CustomModal open={isOpen} onClose={isCloseHandler}>
            <MovieDescription overview={overview}/>
        </CustomModal>
      </div>
    </>
    )
}

ShowInfoButton.propTypes = {
  overview: PropTypes.string,
}

export default ShowInfoButton;