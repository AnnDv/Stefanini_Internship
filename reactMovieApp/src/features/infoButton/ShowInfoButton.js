import React, { useState } from 'react';
import PropTypes from 'prop-types';
import MovieDescription from '../../components/movieComponents/MovieDescription';
import CustomModal from './CustomModal';
import './ShowInfoButton.css';

function ShowInfoButton({ overview }) {
  const [isOpen, setIsOpen] = useState(false);

  const isOpenHandler = () => setIsOpen(true);
  const isCloseHandler = () => setIsOpen(false);

  return (
    <>
      <div className="card-body">
        <button className="button" onClick={isOpenHandler}>View More</button>
        <CustomModal open={isOpen} onClose={isCloseHandler}>
            <MovieDescription data-testid="showInfo" overview={overview}/>
        </CustomModal>
      </div>
    </>
  );
}

ShowInfoButton.propTypes = {
  overview: PropTypes.string,
};

export default ShowInfoButton;
