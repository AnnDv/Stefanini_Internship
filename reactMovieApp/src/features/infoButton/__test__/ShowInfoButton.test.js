import {
  cleanup, render,
} from '@testing-library/react';
import React from 'react';
import ShowInfoButton from '../ShowInfoButton';
import '@testing-library/jest-dom';

describe('Button component', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render a button and have a name', () => {
    const onClick = jest.fn();
    const { getByRole } = render(<ShowInfoButton onClick={onClick} btnTxt="View More"/>);
    const button = getByRole('button');
    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent('View More');
  });
});
