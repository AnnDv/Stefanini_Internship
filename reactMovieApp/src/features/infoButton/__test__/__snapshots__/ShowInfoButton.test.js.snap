// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render a button and have a name 1`] = `
<body>
  <div>
    <div
      class="card-body"
    >
      <button
        class="button"
      >
        View More
      </button>
    </div>
  </div>
</body>
`;
